### DISCUSSION TITLE
tokeio asynchronous function must be executed with 'await'

### DISCUSSION THREAD

**Discussion by zhangxize** [tokio-rs/tokio#6989] [Q&A] [answered]
The tokeio asynchronous function must be executed with 'await'. If it is not executed without 'await', it will never be executed. If it must be executed with 'await', what is the difference between it and a synchronous function? As shown in the following code, the println of the body of the asynchronous function hello-world always fails to execute

async fn hello_world(){
    // thread::sleep(time::Duration::from_secs(2));
    println!("Hello,world66666");
    
}
#[tokio::main]
async fn  main() {
    let future=hello_world();
    // tokio::join!(future);
    // thread::sleep(time::Duration::from_secs(6));
    for i in 1..10000
    {
        println!("Hello, world!");
    }
    let mut buf=String::new();
    io::stdin().read_line(& mut buf);
}

**Comment by Noah-Kennedy** [comment, marked as answer]
Futures in Rust are lazy. An async function or block effectively builds a state machine which is evaluated when awaited (which basically joins the awaited future as a subcomponent of the awaiting future) or spawned on a runtime. Async/await is best thought of as syntactic sugar for building state machines that can be polled to be advanced, and woken for later resumption.

A future is an object representing a computation that can be advanced until it needs to wait for an IO event, at which point it will need to be polled again by a runtime.
